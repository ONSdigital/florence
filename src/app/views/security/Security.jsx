import React from "react";
import PropTypes from "prop-types";
import BackButton from "../../components/back-button";
import Loader from "../../components/loader/Loader";

const Security = ({ signOutAllUsers, openModal, closeModal, loading }) => {
    const popout = {
        id: "sign-out",
        title: "Are you sure you want to sign out all users?",
        body: "Users will need to sign in to Florence again and may lose unsaved changes.",
        buttons: [
            {
                onClick: () => handleSubmit(),
                text: "Sign out all users",
                style: "positive",
            },
            {
                onClick: () => closeModal(["sign-out"]),
                text: "Cancel",
                style: "invert-primary",
            },
        ],
    };

    const handleClick = () => {
        openModal(popout);
    };

    const handleSubmit = () => {
        closeModal(["sign-out"]);
        signOutAllUsers();
    };

    return (
        <div className="grid grid--justify-space-around">
            <div className="grid__col-11 grid__col-md-9">
                <h2 className="margin-top--1">Security</h2>
                <div className="grid">
                    <div className="grid__col-lg-6">
                        {loading ? (
                            <Loader />
                        ) : (
                            <div role="alert" className="panel panel--error panel--no-title margin-top--1 margin-bottom--1">
                                <p className="panel__error">
                                    <strong>Secure Florence</strong>
                                </p>
                                <p>All users will be signed out straight away and will need to sign in again.</p>
                                <button type="submit" className="btn btn--warning margin-top--1 margin-bottom--1" onClick={handleClick}>
                                    Sign out all users
                                </button>
                            </div>
                        )}
                    </div>
                </div>
            </div>
        </div>
    );
};

Security.propTypes = {
    signOutAllUsers: PropTypes.func.isRequired,
    openModal: PropTypes.func.isRequired,
    closeModal: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
};

export default Security;
