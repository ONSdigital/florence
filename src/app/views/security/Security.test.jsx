import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";
import Security from "./Security";

const defaultProps = {
    signOutAllUsers: jest.fn(),
    openModal: jest.fn(),
    closeModal: jest.fn(),
    isLoading: false,
};
const handleSubmit = jest.fn();

describe("Security", () => {
    it("matches the snapshot", () => {
        const tree = renderer.create(<Security {...defaultProps} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("when loading matches the snapshot", () => {
        const props = { ...defaultProps, isLoading: true };
        const tree = renderer.create(<Security {...props} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("requests to open Modal", () => {
        render(<Security {...defaultProps} />);

        userEvent.click(screen.getByText(/sign out all users/i));

        expect(defaultProps.openModal).toBeCalledTimes(1);
        expect(defaultProps.openModal).toBeCalledWith(
            expect.objectContaining({
                body: "Users will need to sign in to Florence again and may lose unsaved changes.",
                id: "sign-out",
                title: "Are you sure you want to sign out all users?",
            })
        );
    });
});
