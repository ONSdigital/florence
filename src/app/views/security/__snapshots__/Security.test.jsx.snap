// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security matches the snapshot 1`] = `
<div
  className="grid grid--justify-space-around"
>
  <div
    className="grid__col-11 grid__col-md-9"
  >
    <h2
      className="margin-top--1"
    >
      Security
    </h2>
    <div
      className="grid"
    >
      <div
        className="grid__col-lg-6"
      >
        <div
          className="panel panel--error panel--no-title margin-top--1 margin-bottom--1"
          role="alert"
        >
          <p
            className="panel__error"
          >
            <strong>
              Secure Florence
            </strong>
          </p>
          <p>
            All users will be signed out straight away and will need to sign in again.
          </p>
          <button
            className="btn btn--warning margin-top--1 margin-bottom--1"
            onClick={[Function]}
            type="submit"
          >
            Sign out all users
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Security when loading matches the snapshot 1`] = `
<div
  className="grid grid--justify-space-around"
>
  <div
    className="grid__col-11 grid__col-md-9"
  >
    <h2
      className="margin-top--1"
    >
      Security
    </h2>
    <div
      className="grid"
    >
      <div
        className="grid__col-lg-6"
      >
        <div
          className="panel panel--error panel--no-title margin-top--1 margin-bottom--1"
          role="alert"
        >
          <p
            className="panel__error"
          >
            <strong>
              Secure Florence
            </strong>
          </p>
          <p>
            All users will be signed out straight away and will need to sign in again.
          </p>
          <button
            className="btn btn--warning margin-top--1 margin-bottom--1"
            onClick={[Function]}
            type="submit"
          >
            Sign out all users
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
